pkg/test/framework/components/echo/kube/instance.go

```diff
@@ -584,3 +586,56 @@ func (c *instance) CallWithRetryOrFail(t test.Failer, opts echo.CallOptions,
 	}
 	return r
 }
+
+func (c *instance) Restart() error {
+	if err := c.Close(); err != nil {
+		return err
+	}
+
+	// fetch original pod count so we can know when the rollout is complete
+	fetch := kubeTest.NewPodMustFetch(c.cluster, c.Config().Namespace.Name(), fmt.Sprintf("%s=%s", "app", c.Config().Service))
+	origPods, err := fetch()
+	if err != nil {
+		return err
+	}
+	if err := c.restartEchoDeployments(); err != nil {
+		return err
+	}
+	// give the rollout a few seconds to get underway
+	time.Sleep(time.Second * 4)
```

Comment:
Why do we need a delay? shouldn't it be eventually consistent with the retry anyways? Do we need to ensure we have a t least one new pod or it will exit too early? If so, we may consider a more robust check?

Reply:
The reasoning behind the delay was that we don't have a great way to distinguish between the pre-rollout and post-rollout state. Going straight from `kubectl rollout` to checking the pod count and health happens too quickly.

Now that I think about it, we could probably just check to make sure pod name(s) changed as a better indicator. I'll give that a try :+1: